optuna_dashboard/_inmemory_cache.py

```diff
@@ -41,6 +39,20 @@ def get_cached_extra_study_property(
         )
 
 
+class InMemoryCache:
+    def __init__(self) -> None:
+        self._cached_extra_study_property_cache: dict[int, "_CachedExtraStudyProperty"] = {}
+        self._cached_extra_study_property_cache_lock = threading.Lock()
+        self._trials_cache: dict[int, list[FrozenTrial]] = {}
+        self._trials_cache_lock = threading.Lock()
+        self._trials_last_fetched_at: dict[int, datetime] = {}
+
+    def clear(self) -> None:
+        self._cached_extra_study_property_cache.clear()
+        self._trials_cache.clear()
+        self._trials_last_fetched_at.clear()
```

Comment:
Do we need the lock here as well?
```suggestion
        with self._cached_extra_study_property_cache_lock:
            self._cached_extra_study_property_cache.clear()
        with self._trials_cache_lock:
            self._trials_cache.clear()
            self._trials_last_fetched_at.clear()
```